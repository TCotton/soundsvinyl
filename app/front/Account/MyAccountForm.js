import { Component } from 'react'
import React from 'react'
import PropTypes from 'prop-types';

export default class MyAccountForm extends Component {

	static propTypes = {
		formName: PropTypes.string.isRequired,
		legend: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		nameLabel: PropTypes.string.isRequired,
		onFormSubmit: PropTypes.func.isRequired,
		onInputChange: PropTypes.func.isRequired,
		password: PropTypes.string.isRequired,
		passwordLabel: PropTypes.string.isRequired
	};

	constructor (props) {
		super(props);

		this.handleSubmitLogin = this.handleSubmitLogin.bind( this );
		this.handleInputChange = this.handleInputChange.bind( this );
	}

	handleSubmitLogin(event) {
		event.preventDefault();

		const { onFormSubmit } = this.props;
		onFormSubmit();
	}

	handleInputChange(event) {
		event.preventDefault();

		const { onInputChange } = this.props;
		onInputChange(event);
	}

	render () {
		const { name, password, legend, nameLabel, passwordLabel, formName } = this.props;

		return (

			<form
				autoComplete='off'
				name={formName}
				onSubmit={this.handleSubmitLogin}
			>
				<fieldset>
					<legend>
						{legend}
					</legend>

					<label htmlFor={`${formName}loginName`}>
						{nameLabel}
					</label>
					<input
						id={`${formName}loginName`}
						name={`${formName}loginName`}
						onChange={this.handleInputChange}
						required
						type='text'
						value={name}
					/>

					<label htmlFor={`${formName}loginPassword`}>
						{passwordLabel}
					</label>
					<input
						id={`${formName}loginPassword`}
						name={`${formName}loginPassword`}
						onChange={this.handleInputChange}
						required
						type='password'
						value={password}
					/>

					<input
						name='loginSubmit'
						type='submit'
						value='Login'
					/>

				</fieldset>
			</form>

		)
	}
}
