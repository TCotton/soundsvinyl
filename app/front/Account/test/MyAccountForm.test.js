import React from 'react'
import MyAccountForm from '../MyAccountForm'
import renderer from 'react-test-renderer'

describe('Component', () => {
	let component
	beforeEach(() => {
		component = renderer.create(
			<MyAccountForm
				disabled
				formName='loginForm'
				legend='Login'
				name='thisIsAName'
				nameLabel='Your email'
				onFormSubmit={() => {}} // eslint-disable-line
				onInputChange={() => {}} // eslint-disable-line
				password='thisIsAPassword'
				passwordLabel='Your password'
			/>
		)
	})

	describe('MyAccountForm', () => {
		it('should be defined', () => {
			expect(component).toBeDefined()
		})

		it('can be updated', () => {
			component.update(
				<MyAccountForm
					disabled={false}
					formName='anotherForm'
					legend='Another Login'
					name='thisIsAnotherName'
					nameLabel='Another email'
					onFormSubmit={() => {}} // eslint-disable-line
					onInputChange={() => {}} // eslint-disable-line
					password='thisIsAnotherPassword'
					passwordLabel='Another password'
				/>
			)

			expect(component).toBeDefined()
		})
	})
})
