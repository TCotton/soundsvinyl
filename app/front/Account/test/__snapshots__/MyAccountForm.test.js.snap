// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MyAccountForm should render correctly 1`] = `
<form
  autoComplete="off"
  name="loginForm"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Login
    </legend>
    <label
      htmlFor="loginFormloginName"
    >
      Your email
    </label>
    <input
      data-autocapitalize="none"
      disabled={true}
      id="loginFormloginName"
      name="loginFormloginName"
      onChange={[Function]}
      required={true}
      type="text"
      value="thisIsAName"
    />
    <label
      htmlFor="loginFormloginPassword"
    >
      Your password
    </label>
    <input
      disabled={true}
      id="loginFormloginPassword"
      name="loginFormloginPassword"
      onChange={[Function]}
      required={true}
      type="password"
      value="thisIsAPassword"
    />
    <input
      name="loginSubmit"
      type="submit"
      value="Login"
    />
  </fieldset>
</form>
`;
